### PR TITLE
[WIP] Port image tests from WAI

### PR DESCRIPTION
**Pull request by tiagoshibata**
**Description**: Port tests and convert data driven tests to googletest

### CODE REVIEW

**Comment by tiagoshibata** [comment]
I'll see if it's fine to keep the tests using brand images

File: winml/test/image/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+* _If bind with Property, what properties should be tested ?, and how to make properties as parameters ?_
+
+### 7. InputPixelFormats
+> We could convert the input image to the following PixelFormats, and then used as input of model. The pixel format expected from model input can be different from the pixel format of real input. But winml can handle the conversion between them.
+
+* Bgra8
+* Rgba8
+* Gray8
+
+# Inplement
```
Comment (ryanlai2) [inline winml/test/image/README.md]: Typo "Implement"

File: winml/test/image/README.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+Basically, we would use Data Driven Testing, which is being used in winmlmodeltests.dll.
+[Table Data Source]( https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/table-data-source)
```
Comment (ryanlai2) [inline winml/test/image/README.md]: Why is Taef mentioned here? I thought Google test is used for data driven testing with testing::WithParamInterface

File: winml/test/image/imagetests.cpp [hunk 3]
```diff
@@ -0,0 +1,1033 @@
...
+        IRandomAccessStream stream = image_file.OpenAsync(FileAccessMode::Read).get();
+        SoftwareBitmap software_bitmap = (BitmapDecoder::CreateAsync(stream).get()).GetSoftwareBitmapAsync().get();
+        VideoFrame frame = VideoFrame::CreateWithSoftwareBitmap(software_bitmap);
+        ImageFeatureValue image_input_tensor = ImageFeatureValue::CreateFromVideoFrame(frame);
+        return image_input_tensor;
+    }
+};
+
+// MNIST model expects 28x28 data with
+class MnistImageTest : public ImageTests, public testing::WithParamInterface<std::pair<std::wstring, unsigned int>> {};
```
Comment (ryanlai2) [inline winml/test/image/imagetests.cpp:317]: Why not use PrepareModelSessionBinding for MnistImageTest?

File: winml/test/image/imagetests.cpp [hunk 4]
```diff
@@ -0,0 +1,1033 @@
...
+    BitmapEncoder encoder = BitmapEncoder::CreateAsync(BitmapEncoder::JpegEncoderId(), write_stream).get();
+    // Set the software bitmap
+    encoder.SetSoftwareBitmap(output_image.SoftwareBitmap());
+
+    encoder.FlushAsync().get();
+
+}
+
+TEST_F(ImageTests, ImageBindingStyleTransfer) {
+    //this test only checks that the operation completed succefully without crashing
```
Comment (ryanlai2) [inline winml/test/image/imagetests.cpp:770]: nit: typo "successfully"

**Pull request merged**